packages/common/container-definitions/src/loader.ts

```diff
@@ -467,6 +472,11 @@ export interface IContainer extends IEventProvider<IContainerEvents> {
 	 * Use this as the primary way of getting access to the user-defined logic within the container.
 	 */
 	getEntryPoint(): Promise<FluidObject>;
+
+	/**
+	 * Exposes any metadata/props related to the container.
+	 */
+	containerMetadata: Record<string, unknown> | undefined;
```

Comment:
Is it undefined for back-compat? Can it be avoided (and use {} if set is empty)?
Or does undefine has some special meaning? Like maybe - we do not know what are the values of all the properties?
If it's the latter, then I assume the previous comment / question compiles to be # 2, i.e. events are overwrites and contain full set of properties.

Reply:
Removed undefined and using {} in case it is empty and not undefined.